xesmf/frontend.py

```diff
@@ -271,6 +272,11 @@ def __init__(
             If not given or if those are not found on the regridded object, regridding
             uses the two last dimensions of the object (or the last one for input LocStreams and Meshes).
 
+        apply_add_nans_to_weights: boolean, optional
+            If True (default), for other than the 'nearest_*' regridding methods, will execute
+            'add_nans_to_weights' to propagate empty weights as NaNs instead of zeros.
+            Target grid cells lying outside of the source domain will then be masked.
```

Comment:
```suggestion
        default_weights: None
            Option to set the default value for undefined weights. By default, ESMF sets this to zero, which leads to 
            extrapolated field having a value of zero instead of being masked. Instead, xESMF sets 
            `default_weight` to `np.nan` for all regridding methods except `nearest_s2d` and `nearest_d2s`. 
            This has for effect to mask target grid cells lying outside of the source domain. 
```

Please explain why we don't need to do this for nearest.

Reply:
I was wondering if it could be useful to set the default weight to something other than 0 or nan, but I finally don't see the point. 

I'm happy with your new parameter description, it clarifies things for me.

Reply:
Ok, thank you. I updated the pull request.

I also do not see any application at the moment to set the default weight to a different value than NaN or 0.
When the aim is to force a certain target cell value other than 0 or NaN, I do not think it is possible with `add_nans_to_weights` alone, as for other values than 0 and NaN also the source cells' values matter when applying the weights. So I guess if one uses a different default weight than 0 or NaN, out-of-source-domain target cells would still be 0, and target cells masked by an output mask but inside the source-domain would have arbitrary values.